internal/clients/http/ec2/ec2_client.go

```diff
@@ -358,3 +365,14 @@ func (c *ec2Client) parseRunInstancesResponse(respAWS *ec2.RunInstancesOutput) [
 	}
 	return list
 }
+
+func (c *ec2Client) GetAccountId(ctx context.Context) (string, error) {
+	input := &sts.GetCallerIdentityInput{}
+
+	out, err := c.sts.GetCallerIdentity(ctx, input)
+	if err != nil {
+		return "", err
+	}
+
+	return *out.Account, nil
+}
```

Comment:
Since you call it from the web process, I would ask you to add telemetry:

```go
	ctx, span := otel.Tracer(TraceName).Start(ctx, "GetAccountId")
	defer span.End()
```

So in case things are slow we know for fact it was AWS API and not our app.